.github/workflows/ci.yml

```diff
@@ -46,19 +46,19 @@ jobs:
         expand:
           - runner: "c7a-2xlarge"
             name: "debug-wasm"
-            cmd: "rustup target add wasm32-unknown-unknown && cargo build --package binius_field --target wasm32-unknown-unknown"
+            cmd: "rustup target add wasm32-unknown-unknown && cargo build --package binius_field --target wasm32-unknown-unknown --features nightly_features"
           - runner: "c7a-2xlarge"
             name: "debug-amd"
-            cmd: "cargo build --tests --benches --examples"
+            cmd: "cargo build --tests --benches --examples --features nightly_features"
           - runner: "c7a-2xlarge"
             name: "debug-amd-no-default-features"
-            cmd: "cargo build --tests --benches --examples --no-default-features"
+            cmd: "cargo build --tests --benches --examples --no-default-features --features nightly_features"
           - runner: "c7a-2xlarge"
             name: "debug-amd-stable"
-            cmd: "cargo +$RUST_VERSION build --tests --benches --examples -p binius_core --features stable_only"
+            cmd: "cargo +$RUST_VERSION build --tests --benches --examples"
           - runner: "c8g-2xlarge"
             name: "debug-arm"
-            cmd: "cargo build --tests --benches --examples"
```

Comment:
In case if enabling `nightly_features` by default it should be smth like `--no-default-features --features rayon`

Reply:
Should `rayon` be completely disabled when building with `--no-default-features`?  Currrently, `binius_maybe_rayon/rayon` is always on, due to `binius_m3` depending on it with that feature enabled.

Reply:
> Currrently, binius_maybe_rayon/rayon is always on, due to binius_m3 depending on it with that feature enabled.

Oh, that seems to be an issue that was introduced recently. The idea was that `rayon` is enabled by default, but when building with `--no-default-featutes` the code should not depend on rayon. But now it seems to be broken. But I propose not to fix it withing the same PR